packages/ibm-products/src/components/UserAvatar/UserAvatar.js

```diff
@@ -0,0 +1,166 @@
+  tooltipText: 'Iam tooltip',
+};
+
+export let UserAvatar = React.forwardRef(
+  (
+    {
+      // The component props, in alphabetical order (for consistency).
+      backgroundColor,
+      className,
+      /* TODO: add other props for UserAvatar, with default values if needed */
+      renderIcon = defaults.renderIcon,
+      tooltipText = defaults.tooltipText,
+      tooltipAlignment = defaults.tooltipAlignment,
+      // Collect any other property values passed in.
+      ...rest
+    },
+    ref
+  ) => {
+    const carbonPrefix = usePrefix();
+    const icons = {
+      user: {
+        md: <User size={32} />,
+      },
+      group: {
+        md: <Group size={32} />,
+      },
+    };
+    const getItem = (renderIcon) => {
+      if (renderIcon === User) {
+        return icons.user['md'];
+      } else if (renderIcon === Group) {
+        return icons.group['md'];
+      } else {
+        return renderIcon;
+      }
+    };
+
+    const SetItem = getItem(renderIcon);
+
+    const renderUserAvatar = () => (
+      <div
+        {
+          // Pass through any other property values as HTML attributes.
+          ...rest
+        }
+        tabIndex={0}
+        className={cx(
+          blockClass, // Apply the block class to the main HTML element
+          className, // Apply any supplied class names to the main HTML element.
+          `${blockClass}--${backgroundColor}`,
+          // example: `${blockClass}__template-string-class-${kind}-n-${size}`,
+          {
+            // switched classes dependant on props or state
+            // example: [`${blockClass}__here-if-small`]: size === 'sm',
+          }
+        )}
+        ref={ref}
+        role="button"
+        {...getDevtoolsProps(componentName)}
+      >
+        <SetItem />
+      </div>
+    );
+
+    return (
+      SetItem &&
+      (tooltipText ? (
+        <Tooltip
+          align={tooltipAlignment}
+          label={tooltipText}
+          className={`${blockClass}__tooltip ${carbonPrefix}--icon-tooltip`}
+        >
+          <TooltipTrigger>{renderUserAvatar()}</TooltipTrigger>
+        </Tooltip>
+      ) : (
+        renderUserAvatar()
+      ))
+    );
+  }
+);
+
+// Return a placeholder if not released and not enabled by feature flag
+UserAvatar = pkg.checkComponentEnabled(UserAvatar, componentName);
+
+// The display name of the component, used by React. Note that displayName
+// is used in preference to relying on function.name.
+UserAvatar.displayName = componentName;
+
+// The types and DocGen commentary for the component props,
+// in alphabetical order (for consistency).
+// See https://www.npmjs.com/package/prop-types#usage.
+UserAvatar.propTypes = {
+  /**
+   * Provide the contents of the UserAvatar.
+   */
+  backgroundColor: PropTypes.oneOf(['light-cyan', 'dark-cyan']),
+
+  /**
+   * Provide an optional class to be applied to the containing node.
+   */
+  className: PropTypes.string,
+  renderIcon: PropTypes.func,
```

Comment:
You have a couple props that are missing comments here, these comments are used as descriptions in the storybook props table.

Reply:
Done the required changes.